### PR TITLE
added example with cmake module inside tool_requires

### PR DESCRIPTION
**Pull request by memsharded**
Docs for https://github.com/conan-io/conan/pull/14264

Can go directly to release/2.0, nothing really new from the release.

### CODE REVIEW

File: examples/graph/tool_requires/use_cmake_modules.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+.. code-block:: cmake
+    :caption: myfunctions/myfunction.cmake
+
+    function(myfunction)
+        message("Hello myfunction!!!!")
+    endfunction()
+
+We can do a ``cd myfunctions && conan create .`` and that will create the ``myfunctions/1.0`` package containing the cmake script.
+
+Then, a consumer package will be like:
```
Comment (RubenRBS) [inline examples/graph/tool_requires/use_cmake_modules.rst:45]:
```suggestion
Then, a consumer package will look like:
```

File: examples/graph/tool_requires/use_cmake_modules.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+And a ``myfunction.cmake`` file in:
+
+.. code-block:: cmake
+    :caption: myfunctions/myfunction.cmake
+
+    function(myfunction)
+        message("Hello myfunction!!!!")
+    endfunction()
+
+We can do a ``cd myfunctions && conan create .`` and that will create the ``myfunctions/1.0`` package containing the cmake script.
```
Comment (RubenRBS) [inline examples/graph/tool_requires/use_cmake_modules.rst:43]:
```suggestion
We can do a ``cd myfunctions && conan create .`` which will create the ``myfunctions/1.0`` package containing the cmake script.
```

**Review by RubenRBS — APPROVED** [review]

**Review by czoido — APPROVED** [review]
I think this should have an analogous PR in the examples2 repo, opening and issue there so we don't forget to add the code and modify the example.

**Comment by memsharded** [comment]
> I think this should have an analogous PR in the examples2 repo, opening and issue there so we don't forget to add the code and modify the example.

I am not fully sure about this (the usage of resources). This is extracted exactly from a test in our test suite. Adding it to examples code seems duplicating the CI for it? It is that necessary? I know I always face the same when doing examples from Conan existing tests... I am good with adding them to the examples2 repo, just something doesn't seem great while doing it.

**Pull request merged**